packages/datatrak-web/src/features/Survey/SurveyContext/SurveyContext.tsx

```diff
@@ -49,14 +52,24 @@ export const SurveyContext = ({ children, surveyCode, countryCode }) => {
   const screenNumber = params.screenNumber ? parseInt(params.screenNumber!, 10) : null;
   const { data: survey } = useSurvey(surveyCode);
   const isResponseScreen = !!urlSearchParams.get('responseId');
+  const isReviewScreen = !!useMatch(ROUTES.SURVEY_REVIEW);
 
-  const { formData } = state;
+  let { formData } = state;
 
   const surveyScreens = survey?.screens || [];
   const flattenedScreenComponents = getAllSurveyComponents(surveyScreens);
   const primaryEntityQuestion = flattenedScreenComponents.find(
     question => question.type === QuestionType.PrimaryEntity,
   );
+  const autoFillAnswers = usePrimaryEntityQuestionAutoFill(
+    primaryEntityQuestion,
+    flattenedScreenComponents,
+    primaryEntityCode,
+  );
```

Comment:
What will happen when the entity ancestors are loading, if `initialiseFormData` is only getting called on `surveyCode` change?

Reply:
This is actually outside of initialiseFormData so when the ancestors are finished loading, the data will just get added to the formData